### DISCUSSION TITLE
JS Admin Auth createUser documentation confuses/misleading

### DISCUSSION THREAD

**Discussion by Logaritma26** [supabase/supabase#20441] [Contribute to Supabase]
# Improve documentation

## Link 

[Admin auth createUser](https://supabase.com/docs/reference/javascript/auth-admin-createuser)

## Describe the problem

Its confuse a bit because first time when I see it, it was similar to normal user signUp but in admin mode. So I was expecting to receive confirmation email after I create user. But from what I learned after discussion in DC channel it actually not. People may confuse like me.

## Describe the improvement

Putting a little note below the description about this is not same as signUp user and will not send confirmation mail to user may help.

**Comment by FatumaA** [comment]
It does not auto send confirmation email but it says you can add that to the function here:

<img width="1138" alt="Screenshot" src="https://github.com/supabase/supabase/assets/67555014/06dc7a3b-5f97-4585-b654-0d3501ad64b3">

**Comment by gregnr** [comment]
Thanks for the issue @Logaritma26. @FatumaA has a good point - between the note on the left and the `Auto-confirm the user's email` example on the right, it feels like the docs cover this well enough. Let me know if you disagree.

**Comment by Logaritma26** [comment]
The problem is, if you pass email_confirm false, then it does not send a confirmation mail. When i saw this param, i thought it should send confirmation mail. Thats the confusing part.

**Comment by FatumaA** [comment]
Ah I get you.
I think normally booleans start off as false unless explicitly set to true

**Comment by gregnr** [comment]
@Logaritma26 thanks for explaining. Just so that I understand, you were expecting `email_confirm` to be `true` by default? My understanding is that since this is an admin-focused operation, we don't want to make any assumptions about sending emails (but still give the option to).

I agree this flag should be as clear as possible. What would you suggest we change to make this clearer? Note we do mention in the left description that this is `false` by default.

**Comment by Logaritma26** [comment]
I hope this wouldn't be this hard to explain :D. I wasn't expecting it to be `true` by default. I was expecting when this field is selected `false` (which is default and thats completely OK this way), then if I create a user this way, that user should get a confirmation mail because I selected `email_confirm` false so its not going to auto confirm email and user should be able to confirm it ? The problem is there is no way user can confirm his account, if the account created by admin auth client and `email_confirm` is false. Let me know if this explains or you need more context

**Comment by GaryAustin1** [comment]
This is confusing because the way this flag works in other places it actually does something.
It took me a while helping the OP to realize the flag does nothing for this operation versus SignUp where it changes whether you send a confirm or not.  Discord: https://discord.com/channels/839993398554656828/1107340805523382403/1107734127156924436

I spent time replicating and testing only to find out it does nothing.

The flag should not be present at all, and it is not clear it is even in the source code as being passed on, but only did a quick glance.

**Comment by gregnr** [comment]
Thanks for the extra details @Logaritma26, definitely clears up the confusion. Thanks @GaryAustin1 for the extra context. cc @supabase/auth who might have some insight on the naming/logic.

**Comment by J0** [comment]
Hey all,

Thanks for the discussion and for taking the time to explain the issue. The source of confusion might come from the fact that the `email_confirmed` param [determines whether the created user is confirmed](https://github.com/supabase/gotrue/blob/master/internal/api/admin.go#L418) .  This might be slightly confusing as developers might be expecting `email_confirmed` to toggle whether a confirmation is sent since we do have a similar `Autoconfirm` option on our dashboard which allows developers to toggle whether or not a confirmation email is sent.


We'll get discuss and get back to you in a bit

Joel
